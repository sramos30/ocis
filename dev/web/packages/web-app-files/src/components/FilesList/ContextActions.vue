<template>
  <context-action-menu :menu-sections="menuSections" :items="items" />
</template>

<script lang="ts">
import ContextActionMenu from '../ContextActionMenu.vue'

import FileActions from '../../mixins/fileActions'
import AcceptShare from '../../mixins/actions/acceptShare'
import Copy from '../../mixins/actions/copy'
import CreatePublicLink from '../../mixins/actions/createPublicLink'
import DeclineShare from '../../mixins/actions/declineShare'
import Delete from '../../mixins/actions/delete'
import DownloadArchive from '../../mixins/actions/downloadArchive'
import DownloadFile from '../../mixins/actions/downloadFile'
import EmptyTrashBin from '../../mixins/actions/emptyTrashBin'
import Favorite from '../../mixins/actions/favorite'
import Move from '../../mixins/actions/move'
import Navigate from '../../mixins/actions/navigate'
import Rename from '../../mixins/actions/rename'
import Restore from '../../mixins/actions/restore'
import ShowActions from '../../mixins/actions/showActions'
import ShowDetails from '../../mixins/actions/showDetails'
import ShowShares from '../../mixins/actions/showShares'
import SetSpaceImage from '../../mixins/spaces/actions/setImage'
import SetSpaceReadme from '../../mixins/spaces/actions/setReadme'
import SpaceNavigate from '../../mixins/spaces/actions/navigate'
import { PropType } from '@vue/composition-api'
import { Resource } from '../../helpers/resource'

export default {
  name: 'ContextActions',
  components: { ContextActionMenu },
  mixins: [
    FileActions,
    AcceptShare,
    Copy,
    CreatePublicLink,
    DeclineShare,
    Delete,
    DownloadArchive,
    DownloadFile,
    EmptyTrashBin,
    Favorite,
    Move,
    Navigate,
    Rename,
    Restore,
    ShowActions,
    ShowDetails,
    ShowShares,
    SetSpaceImage,
    SetSpaceReadme,
    SpaceNavigate
  ],

  props: {
    items: {
      type: Array as PropType<Resource[]>,
      required: true
    }
  },

  computed: {
    menuSections() {
      const sections = []
      if (this.items.length > 1) {
        if (this.menuItemsBatchActions.length) {
          sections.push({
            name: 'batch-actions',
            items: [...this.menuItemsBatchActions]
          })
        }
        sections.push({
          name: 'batch-details',
          items: [...this.$_showDetails_items]
        })
        return sections
      }

      if (this.menuItemsContext.length) {
        sections.push({
          name: 'context',
          items: this.menuItemsContext
        })
      }
      if (this.menuItemsShare.length) {
        sections.push({
          name: 'share',
          items: this.menuItemsShare
        })
      }
      if (this.menuItemsActions.length) {
        sections.push({
          name: 'actions',
          items: this.menuItemsActions
        })
      }
      if (this.menuItemsSidebar.length) {
        sections.push({
          name: 'sidebar',
          items: this.menuItemsSidebar
        })
      }
      return sections
    },

    filterParams() {
      return {
        resources: this.items
      }
    },

    menuItemsBatchActions() {
      return [
        ...this.$_acceptShare_items,
        ...this.$_declineShare_items,
        ...this.$_downloadArchive_items,
        ...this.$_delete_items,
        ...this.$_move_items,
        ...this.$_copy_items,
        ...this.$_emptyTrashBin_items,
        ...this.$_restore_items
      ].filter((item) => item.isEnabled(this.filterParams))
    },

    menuItemsContext() {
      const fileHandlers = [
        ...this.$_fileActions_editorActions,
        ...this.$_fileActions_loadExternalAppActions(this.filterParams.resources)
      ]

      return [...fileHandlers].filter((item) => item.isEnabled(this.filterParams))
    },

    menuItemsShare() {
      return [...this.$_showShares_items, ...this.$_createPublicLink_items].filter((item) =>
        item.isEnabled(this.filterParams)
      )
    },

    menuItemsActions() {
      return [
        ...this.$_downloadArchive_items,
        ...this.$_downloadFile_items,
        ...this.$_delete_items,
        ...this.$_move_items,
        ...this.$_copy_items,
        ...this.$_rename_items,
        ...this.$_restore_items,
        ...this.$_acceptShare_items,
        ...this.$_declineShare_items,
        ...this.$_setSpaceImage_items,
        ...this.$_setSpaceReadme_items,
        ...this.$_navigate_space_items
      ].filter((item) => item.isEnabled(this.filterParams))
    },

    menuItemsSidebar() {
      const fileHandlers = [...this.$_navigate_items]

      return [
        ...this.$_favorite_items.map((action) => {
          action.keepOpen = true
          return action
        }),
        ...fileHandlers,
        ...this.$_showDetails_items
      ].filter((item) => item.isEnabled(this.filterParams))
    }
  }
}
</script>
