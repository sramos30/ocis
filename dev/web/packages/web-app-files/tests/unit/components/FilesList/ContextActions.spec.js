import Vuex from 'vuex'
import DesignSystem from 'owncloud-design-system'
import stubs from '@/tests/unit/stubs/index.js'
import { createLocalVue, mount } from '@vue/test-utils'
import ContextActions from '../../../../src/components/FilesList/ContextActions.vue'
import GetTextPlugin from 'vue-gettext'
import fixtureMimeTypes from '@files/tests/__fixtures__/mimeTypes.js'

const localVue = createLocalVue()
localVue.use(DesignSystem)
localVue.use(Vuex)
localVue.use(GetTextPlugin, {
  translations: 'does-not-matter.json',
  silent: true
})

const exampleApps = [
  { name: 'exampleName1', icon: 'exampleIcon1' },
  { name: 'exampleName2', icon: 'exampleIcon2' },
  { name: 'exampleName3', icon: 'exampleIcon3' }
]

const mockMenuSections = [
  {
    name: 'context',
    items: [
      {
        icon: 'file-text',
        canBeDefault: true,
        handler: jest.fn(),
        label: () => 'Open in editor',
        componentType: 'oc-button',
        selector: '.oc-files-actions-markdown-editor-trigger',
        class: 'oc-files-actions-markdown-editor-trigger'
      },
      ...exampleApps.map((app) => {
        return {
          img: app.icon,
          canBeDefault: true,
          handler: jest.fn(),
          label: () => 'Open in ' + app.name,
          componentType: 'oc-button',
          selector: `.oc-files-actions-${app.name}-trigger`,
          class: `oc-files-actions-${app.name}-trigger`
        }
      }),
      {
        icon: 'download',
        canBeDefault: true,
        handler: jest.fn(),
        label: () => 'Download',
        componentType: 'oc-button',
        selector: '.oc-files-actions-download-file-trigger',
        class: 'oc-files-actions-download-file-trigger'
      },
      {
        icon: 'links',
        handler: jest.fn(),
        label: () => 'Create link',
        componentType: 'oc-button',
        selector: '.oc-files-actions-create-public-link-trigger',
        class: 'oc-files-actions-create-public-link-trigger'
      },
      {
        icon: 'group',
        handler: jest.fn(),
        label: () => 'Share',
        componentType: 'oc-button',
        selector: '.oc-files-actions-show-shares-trigger',
        class: 'oc-files-actions-show-shares-trigger'
      }
    ]
  },
  {
    name: 'actions',
    items: [
      {
        icon: 'pencil',
        handler: jest.fn(),
        label: () => 'Edit',
        componentType: 'oc-button',
        selector: '.oc-files-actions-rename-trigger',
        class: 'oc-files-actions-rename-trigger'
      },
      {
        icon: 'folder-shared',
        handler: jest.fn(),
        label: () => 'Move',
        componentType: 'oc-button',
        selector: '.oc-files-actions-move-trigger',
        class: 'oc-files-actions-move-trigger'
      },
      {
        icon: 'file-copy',
        handler: jest.fn(),
        label: () => 'Copy',
        componentType: 'oc-button',
        selector: '.oc-files-actions-copy-trigger',
        class: 'oc-files-actions-copy-trigger'
      },
      {
        icon: 'delete-bin-5',
        handler: jest.fn(),
        label: () => 'Delete',
        componentType: 'oc-button',
        selector: '.oc-files-actions-delete-trigger',
        class: 'oc-files-actions-delete-trigger'
      },
      {
        icon: 'slideshow',
        handler: jest.fn(),
        label: () => 'All actions',
        componentType: 'oc-button',
        selector: '.oc-files-actions-show-actions-trigger',
        class: 'oc-files-actions-show-actions-trigger'
      }
    ]
  },
  {
    name: 'sidebar',
    items: [
      {
        icon: 'information',
        handler: jest.fn(),
        label: () => 'Details',
        componentType: 'oc-button',
        selector: '.oc-files-actions-show-details-trigger',
        class: 'oc-files-actions-show-details-trigger'
      }
    ]
  }
]

const filesPersonalRoute = { name: 'files-personal' }

describe('ContextActions', () => {
  describe('action handlers', () => {
    afterEach(() => {
      jest.clearAllMocks()
    })
    it('renders action handlers as clickable elements', async () => {
      const wrapper = getWrapper(
        filesPersonalRoute,
        {
          name: 'exampleFile',
          extension: 'jpg',
          mimeType: 'application/fileFormat2',
          type: 'file'
        },
        exampleApps
      )

      for (const section of mockMenuSections) {
        for (const item of section.items) {
          const buttonElement = wrapper.find(item.selector)
          expect(buttonElement.exists()).toBeTruthy()
          await buttonElement.trigger('click')
          expect(item.handler).toHaveBeenCalledTimes(1)
        }
      }
    })
  })

  describe('menu items', () => {
    it('renders a list of actions for a file', () => {
      const wrapper = getWrapper(filesPersonalRoute, {
        name: 'exampleFile',
        extension: 'jpg',
        mimeType: 'application/fileFormat2',
        type: 'file'
      })

      expect(wrapper).toMatchSnapshot()
    })

    it('renders a list of actions for a folder', () => {
      const wrapper = getWrapper(filesPersonalRoute, {
        name: 'exampleFolder',
        extension: '',
        type: 'folder'
      })

      expect(wrapper).toMatchSnapshot()
    })
  })
})

function getWrapper(route, { filename, extension, type = '', mimeType }, availableApps = []) {
  const mountStubs = { ...stubs, 'oc-button': false }

  return mount(ContextActions, {
    localVue,
    store: createStore(),
    data() {
      return {
        appList: availableApps
      }
    },
    stubs: mountStubs,
    mocks: {
      $route: route,
      publicPage: () => false
    },
    provide: {
      currentSpace: {
        value: {}
      }
    },
    propsData: {
      items: [
        {
          id: 'a93f8adf==',
          fileId: 'a93f8adf==',
          name: filename,
          path: type === 'file' ? `/${filename}.${extension}` : `/${filename}`,
          mimeType,
          extension,
          type,
          canDownload: () => true,
          isReceivedShare: () => true,
          canBeDeleted: () => true,
          canRename: () => true
        }
      ]
    },
    computed: {
      menuSections: () => mockMenuSections
    }
  })
}

function createStore(state) {
  return new Vuex.Store({
    getters: {
      capabilities: jest.fn(() => ({
        files: {
          app_providers: [
            {
              apps_url: '/app/list',
              enabled: true,
              open_url: '/app/open'
            }
          ]
        }
      }))
    },
    modules: {
      External: {
        state: {
          ...state
        },
        namespaced: true,
        getters: {
          mimeTypes: () => {
            return fixtureMimeTypes
          }
        }
      },
      Files: {
        state: {
          ...state
        },
        namespaced: true,
        getters: {
          currentFolder: () => '/'
        }
      }
    }
  })
}
