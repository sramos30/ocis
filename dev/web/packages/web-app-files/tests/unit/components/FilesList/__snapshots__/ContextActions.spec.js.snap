// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextActions menu items renders a list of actions for a file 1`] = `
<div id="oc-files-context-menu">
  <oc-list-stub id="oc-files-context-actions-context" class="oc-files-context-actions oc-pb-s oc-files-context-actions-border">
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-markdown-editor-trigger action-menu-item" data-testid="action-handler">
        <oc-icon-stub name="file-text" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color="" data-testid="action-icon"></oc-icon-stub> <span data-testid="action-label" class="oc-files-context-action-label">Open in editor</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-exampleName1-trigger action-menu-item" data-testid="action-handler">
        <oc-img-stub src="exampleIcon1" alt="" loadingtype="eager" data-testid="action-img" class="oc-icon oc-icon-m"></oc-img-stub> <span data-testid="action-label" class="oc-files-context-action-label">Open in exampleName1</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-exampleName2-trigger action-menu-item" data-testid="action-handler">
        <oc-img-stub src="exampleIcon2" alt="" loadingtype="eager" data-testid="action-img" class="oc-icon oc-icon-m"></oc-img-stub> <span data-testid="action-label" class="oc-files-context-action-label">Open in exampleName2</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-exampleName3-trigger action-menu-item" data-testid="action-handler">
        <oc-img-stub src="exampleIcon3" alt="" loadingtype="eager" data-testid="action-img" class="oc-icon oc-icon-m"></oc-img-stub> <span data-testid="action-label" class="oc-files-context-action-label">Open in exampleName3</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-download-file-trigger action-menu-item" data-testid="action-handler">
        <oc-icon-stub name="download" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color="" data-testid="action-icon"></oc-icon-stub> <span data-testid="action-label" class="oc-files-context-action-label">Download</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-create-public-link-trigger action-menu-item" data-testid="action-handler">
        <oc-icon-stub name="links" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color="" data-testid="action-icon"></oc-icon-stub> <span data-testid="action-label" class="oc-files-context-action-label">Create link</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-show-shares-trigger action-menu-item" data-testid="action-handler">
        <oc-icon-stub name="group" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color="" data-testid="action-icon"></oc-icon-stub> <span data-testid="action-label" class="oc-files-context-action-label">Share</span>
        <!---->
      </button></li>
  </oc-list-stub>
  <oc-list-stub id="oc-files-context-actions-actions" class="oc-files-context-actions oc-pb-s oc-pt-s oc-files-context-actions-border">
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-rename-trigger action-menu-item" data-testid="action-handler">
        <oc-icon-stub name="pencil" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color="" data-testid="action-icon"></oc-icon-stub> <span data-testid="action-label" class="oc-files-context-action-label">Edit</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-move-trigger action-menu-item" data-testid="action-handler">
        <oc-icon-stub name="folder-shared" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color="" data-testid="action-icon"></oc-icon-stub> <span data-testid="action-label" class="oc-files-context-action-label">Move</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-copy-trigger action-menu-item" data-testid="action-handler">
        <oc-icon-stub name="file-copy" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color="" data-testid="action-icon"></oc-icon-stub> <span data-testid="action-label" class="oc-files-context-action-label">Copy</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-delete-trigger action-menu-item" data-testid="action-handler">
        <oc-icon-stub name="delete-bin-5" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color="" data-testid="action-icon"></oc-icon-stub> <span data-testid="action-label" class="oc-files-context-action-label">Delete</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-show-actions-trigger action-menu-item" data-testid="action-handler">
        <oc-icon-stub name="slideshow" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color="" data-testid="action-icon"></oc-icon-stub> <span data-testid="action-label" class="oc-files-context-action-label">All actions</span>
        <!---->
      </button></li>
  </oc-list-stub>
  <oc-list-stub id="oc-files-context-actions-sidebar" class="oc-files-context-actions oc-pt-s">
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-show-details-trigger action-menu-item" data-testid="action-handler">
        <oc-icon-stub name="information" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color="" data-testid="action-icon"></oc-icon-stub> <span data-testid="action-label" class="oc-files-context-action-label">Details</span>
        <!---->
      </button></li>
  </oc-list-stub>
</div>
`;

exports[`ContextActions menu items renders a list of actions for a folder 1`] = `
<div id="oc-files-context-menu">
  <oc-list-stub id="oc-files-context-actions-context" class="oc-files-context-actions oc-pb-s oc-files-context-actions-border">
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-markdown-editor-trigger action-menu-item" data-testid="action-handler">
        <oc-icon-stub name="file-text" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color="" data-testid="action-icon"></oc-icon-stub> <span data-testid="action-label" class="oc-files-context-action-label">Open in editor</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-exampleName1-trigger action-menu-item" data-testid="action-handler">
        <oc-img-stub src="exampleIcon1" alt="" loadingtype="eager" data-testid="action-img" class="oc-icon oc-icon-m"></oc-img-stub> <span data-testid="action-label" class="oc-files-context-action-label">Open in exampleName1</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-exampleName2-trigger action-menu-item" data-testid="action-handler">
        <oc-img-stub src="exampleIcon2" alt="" loadingtype="eager" data-testid="action-img" class="oc-icon oc-icon-m"></oc-img-stub> <span data-testid="action-label" class="oc-files-context-action-label">Open in exampleName2</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-exampleName3-trigger action-menu-item" data-testid="action-handler">
        <oc-img-stub src="exampleIcon3" alt="" loadingtype="eager" data-testid="action-img" class="oc-icon oc-icon-m"></oc-img-stub> <span data-testid="action-label" class="oc-files-context-action-label">Open in exampleName3</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-download-file-trigger action-menu-item" data-testid="action-handler">
        <oc-icon-stub name="download" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color="" data-testid="action-icon"></oc-icon-stub> <span data-testid="action-label" class="oc-files-context-action-label">Download</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-create-public-link-trigger action-menu-item" data-testid="action-handler">
        <oc-icon-stub name="links" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color="" data-testid="action-icon"></oc-icon-stub> <span data-testid="action-label" class="oc-files-context-action-label">Create link</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-show-shares-trigger action-menu-item" data-testid="action-handler">
        <oc-icon-stub name="group" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color="" data-testid="action-icon"></oc-icon-stub> <span data-testid="action-label" class="oc-files-context-action-label">Share</span>
        <!---->
      </button></li>
  </oc-list-stub>
  <oc-list-stub id="oc-files-context-actions-actions" class="oc-files-context-actions oc-pb-s oc-pt-s oc-files-context-actions-border">
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-rename-trigger action-menu-item" data-testid="action-handler">
        <oc-icon-stub name="pencil" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color="" data-testid="action-icon"></oc-icon-stub> <span data-testid="action-label" class="oc-files-context-action-label">Edit</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-move-trigger action-menu-item" data-testid="action-handler">
        <oc-icon-stub name="folder-shared" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color="" data-testid="action-icon"></oc-icon-stub> <span data-testid="action-label" class="oc-files-context-action-label">Move</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-copy-trigger action-menu-item" data-testid="action-handler">
        <oc-icon-stub name="file-copy" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color="" data-testid="action-icon"></oc-icon-stub> <span data-testid="action-label" class="oc-files-context-action-label">Copy</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-delete-trigger action-menu-item" data-testid="action-handler">
        <oc-icon-stub name="delete-bin-5" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color="" data-testid="action-icon"></oc-icon-stub> <span data-testid="action-label" class="oc-files-context-action-label">Delete</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-show-actions-trigger action-menu-item" data-testid="action-handler">
        <oc-icon-stub name="slideshow" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color="" data-testid="action-icon"></oc-icon-stub> <span data-testid="action-label" class="oc-files-context-action-label">All actions</span>
        <!---->
      </button></li>
  </oc-list-stub>
  <oc-list-stub id="oc-files-context-actions-sidebar" class="oc-files-context-actions oc-pt-s">
    <li class="oc-files-context-action oc-px-s oc-rounded"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-show-details-trigger action-menu-item" data-testid="action-handler">
        <oc-icon-stub name="information" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color="" data-testid="action-icon"></oc-icon-stub> <span data-testid="action-label" class="oc-files-context-action-label">Details</span>
        <!---->
      </button></li>
  </oc-list-stub>
</div>
`;
